adserver/management/commands/archive_offers.py

```diff
@@ -0,0 +1,229 @@
+"""
+Archives old offers to CSV files.
+
+The offers table can get very large
+and it's important for performance to keep it as small as possible.
+As a result, archiving old offers can have a good effect on performance.
+This management command archives old offers to CSV files, gzips them,
+can copy them to remote storage (settings.BACKUPS_STORAGE)
+and with a passed flag can delete the archives from the DB.
+"""
+import datetime
+import subprocess
+import tempfile
+from pathlib import Path
+
+from django.conf import settings
+from django.core.files.storage import get_storage_class
+from django.core.management import CommandError
+from django.core.management.base import BaseCommand
+from django.db import connection
+from django.utils import timezone
+from django.utils.translation import ugettext_lazy as _
+
+
+class Command(BaseCommand):
+
+    """Management command to help archive offers."""
+
+    help = "Archives offers in CSV form from the database to files: one per day."
+
+    default_output_dir = "/tmp"
+    default_end_date = (timezone.now() - datetime.timedelta(days=90)).date()
+    default_start_date = default_end_date - datetime.timedelta(days=30)
```

Comment:
Are we thinking of running this nightly now? Could be good to keep the DB performance consistent.

Reply:
I was still thinking maybe weekly or monthly. It shouldn't take nearly as much IO to do it day by day.